webtau-http/src/main/java/com/twosigma/webtau/http/Http.java

```diff
@@ -730,6 +877,23 @@ private HttpResponse request(String method, String fullUrl,
         }
     }
 
+    private void setRequestMethod(String method, HttpURLConnection connection) throws ProtocolException {
+        if (method.equals("PATCH")) {
```

Comment:
This is very sad :(

Reply:
indeed. It gets slightly hackier if you need to support `patch` over https.
I guess this is Oracle's way to suggest moving up to the latest LTS (11)

Reply:
Oh, so you're suggesting this approach won't work for PATCH https?  If that is the case I would suggest we fail loudly and explicitly, presumably right now we'll fail in weird and confusing ways.

Reply:
I think I can make it work for https, or otherwise make it scream in some obvious way.

Reply:
If you need this quickly for http then let's scream but if it can wait for https support then I think we should do that.  Essentially, I'm saying up to you but I think we should do one or the other, the current approach will likely fail in really non-obvious ways (correct me if I'm wrong).

Reply:
We use https, so this needs to be working.

I just checked in the https work-around. I kept all the hacky stuff in a single method although it violates my personal preference for "deep nesting avoidance". Let me know what you think.